products/views.py

```diff
@@ -1,3 +1,63 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http      import JsonResponse
+from django.views     import View
+from django.db.models import Q
+
+from datetime       import datetime
+from .models        import Product, Efficacy, ProductEfficacy, ProductSummary
+
+class ProductsView(View):
+    def get(self, request):
+        try:          
+            efficacy = request.GET.getlist('efficacy',None)
+            q_object = Q()
+            if efficacy:
+                for i in efficacy:                    
+                    q_object.add(Q(Q(efficacy__id=i)),Q.OR)
```

Comment:
여기서 굳이 반복문을 사용하지 않아도 처리가 가능합니다.
```suggestion
    q = Q()
    ...
    q &= Q(~~~~~ = efficacy)
    ...
    ~~~.objects.filter(q)
```

Reply:
수정완료했습니다!